### PR TITLE
Bugfix/search for old pegins

### PR DESCRIPTION
**Pull request by julia-zack**
The condition to consider a tx a pegin is changed: instead of compare the transaction's destination address to the federation address  -that was causing trouble because of the new federation-, we are using the "isBtcTxHashAlreadyProcessed" method from the bridge.

I had to change some tests too because they used the old condition.

### CODE REVIEW

File: src/services/pegin-status/pegin-status.service.ts [hunk 1]
```diff
@@ -157,15 +159,18 @@ export class PeginStatusService {
...
-  private isSentToFederationAddress(federationAddress: string, vout: Vout[]): boolean {
-    let found = false;
-    for (let i = 0; vout && i < vout.length && !found; i++) {
-      if (federationAddress === vout[i].addresses[0]) {
-        found = true;
-      }
-    }
-    return found;
-  }
+
```
Comment (alexjavabraz) [inline src/services/pegin-status/pegin-status.service.ts:162]: You can remove this commented code.

File: src/services/pegin-status/pegin-status.service.ts [hunk 2]
```diff
@@ -110,10 +110,12 @@ export class PeginStatusService {
           this.logger.debug(errorMessage);
           this.status = Status.ERROR_NOT_A_PEGIN;
         } else {
+
           const federationAddress = await this.bridgeService.getFederationAddress();
```
Comment (alexjavabraz) [inline src/services/pegin-status/pegin-status.service.ts:114]: federationAddress is unnecessary in this context, please remove it.

Comment (julia-zack) [reply src/services/pegin-status/pegin-status.service.ts:114]: is used just after! in line 125. but i agree i can move it from line 114 to line 122

**Review by alexjavabraz — APPROVED** [review]
LGTM

File: src/__tests__/unit/services/pegin-status.service.unit.ts [hunk 3]
```diff
@@ -48,10 +48,14 @@ const getRskInfo = (btcTxId: string) => {
   return data;
 }
 
+
+// not sure if have to change some errors tests and set
+// isTxProcessed to false (or true)
```
Comment (ronaldsg20) [inline src/__tests__/unit/services/pegin-status.service.unit.ts:53]: Yes that should be the testing approach

File: src/__tests__/unit/bridge.handler.unit.ts [hunk 4]
```diff
@@ -22,6 +24,15 @@ describe('Service: Bridge', () => {
     expect(lockingCap).to.be.Number();
   });
 
+  it('returns true if tx hash was processed by bridge, false if not', async () => {
```
Comment (ronaldsg20) [inline src/__tests__/unit/bridge.handler.unit.ts:27]: Please add a test using a valid btc tx for pegin sent to a previous Federation address

**Review by ronaldsg20 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**